lib/components/table/table.ts

```diff
@@ -145,54 +126,124 @@ export class TableController extends Stacks.StacksController {
 
         // update the UI and set the `data-sort-direction` attribute if appropriate, so that the next click
         // will cause sorting in descending direction
-        this.setCurrentSort(colHead, direction === 1 ? "asc" : "desc");
+        this.updateSortedColumnStyles(
+            colHead,
+            direction === 1 ? SortOrder.Ascending : SortOrder.Descending
+        );
     }
+
+    private updateSortedColumnStyles = (
+        targetColumnHeader: Element,
+        direction: SortOrder
+    ): void => {
+        // Loop through all sortable columns and remove their sorting direction
+        // (if any), and only leave/set a sorting on `targetColumnHeader`.
+        this.columnTargets.forEach((header: HTMLTableCellElement) => {
+            const isCurrent = header === targetColumnHeader;
+            const classSuffix = isCurrent
+                ? direction === SortOrder.Ascending
+                    ? "asc"
+                    : "desc"
+                : SortOrder.None;
+
+            header.classList.toggle(
+                "is-sorted",
+                isCurrent && direction !== SortOrder.None
+            );
+            header.querySelectorAll(".js-sorting-indicator").forEach((icon) => {
+                icon.classList.toggle(
+                    "d-none",
+                    !icon.classList.contains(
+                        "js-sorting-indicator-" + classSuffix
+                    )
+                );
+            });
+
+            if (isCurrent) {
+                header.setAttribute("aria-sort", direction);
+            } else {
+                header.removeAttribute("aria-sort");
+            }
+        });
+    };
+
+    /**
+     * Transform legacy header markup into the new markup.
+     *
+     * @param headerEl
+     */
+    private ensureHeadersAreClickable = (headerEl: HTMLTableCellElement) => {
```

Comment:
I'm trying to recall my thoughts from when we discussed this a couple weeks ago 😂 Feels like forever ago. Anyways…

> I feel strongly about NOT introducing a permanent markup transformer in Stacks to support legacy usage.
If we can automate the change at runtime we can do this also with a codemod that run once in Core before the library get upgraded.
We could look into [codemods](https://github.com/rajasegar/awesome-codemods) to automate everything but I suspect it will go faster to semi-automatically adjust (e.g. replace with regex in editor) the [20 occurrences](https://github.com/search?q=repo%3AStackEng%2FStackOverflow+click-%3Es-table%23sort&type=code) in Core.

I agree. I think the markup transformer is a little too much magic for Stacks. I _think_ we could do the find+replace without too much complication. Only big issue I could see is if any tables are partially rendered with JS, but I figure we could always open a PR and find out.

>  I don't feel strongly about following semantic versioning by the book honestly

 As for the semver discussion, I remember starting from a place of "I want to follow semver strictly" to "strict semver can cause complication for a library like ours". I doubt we'd benefit from waiting for several breaking changes as the overhead for our small team would be a lot, so we'd instead be doing major releases for any minor breaking change. So, I'm good to ship the occasional breaking change in minor versions as long as we communicate it and provide support.